### PR TITLE
aaronmassey/emul sbs gauge reg rw improvements

### PR DESCRIPTION
**Pull request by aaronemassey**
Emulator improvements that makeup the first two commits of https://github.com/zephyrproject-rtos/zephyr/pull/46817

Once this merges, I'll rebase that PR's branch on top of main so make it a smaller PR.

### CODE REVIEW

File: subsys/emul/i2c/emul_sbs_gauge.c [hunk 1]
```diff
@@ -102,7 +104,7 @@ static int sbs_gauge_emul_transfer_i2c(const struct emul *target, struct i2c_msg
 		if (msgs->flags & I2C_MSG_READ) {
 			switch (msgs->len - 1) {
 			case 1:
-				val = reg_read(target, reg);
+				rc = emul_sbs_gauge_reg_read(target, reg, &val);
 				msgs->buf[0] = val;
```
Comment (yperess) [inline subsys/emul/i2c/emul_sbs_gauge.c:108]: I think this needs to check `rc` first

**Review by yperess — CHANGES_REQUESTED** [review]

File: subsys/emul/i2c/emul_sbs_gauge.c [hunk 2]
```diff
@@ -61,15 +62,15 @@ static int reg_read(const struct emul *target, int reg)
...
 	case SBS_GAUGE_CMD_DESIGN_VOLTAGE:
 		/* Arbitrary stub value. */
-		val = 1;
+		*val = 1;
 		break;
 	default:
 		LOG_ERR("Unknown register 0x%x read", reg);
 		return -EIO;
 	}
 	LOG_INF("read 0x%x = 0x%x", reg, val);
```
Comment (fabiobaltieri) [inline subsys/emul/i2c/emul_sbs_gauge.c:71]:
```
/__w/zephyr/zephyr/subsys/emul/i2c/emul_sbs_gauge.c: In function 'emul_sbs_gauge_reg_read':
/__w/zephyr/zephyr/subsys/emul/i2c/emul_sbs_gauge.c:71:17: error: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'int *' [-Werror=format=]
   71 |         LOG_INF("read 0x%x = 0x%x", reg, val);
      |                 ^~~~~~~~~~~~~~~~~~       ~~~
      |                                          |
      |                                          int *
```

**Review by yperess — DISMISSED** [review]

File: subsys/emul/i2c/emul_sbs_gauge.c (hunk 2 above)
Comment (aaronemassey) [reply subsys/emul/i2c/emul_sbs_gauge.c]: Fixed, but for some reason I didn't get this build failure locally. Do you have yours setup with `clang-tidy`?

**Review by alevkoy — DISMISSED** [review]
The function renames look fine to me, but could they actually be overridden by other symbols before? They appear to be strongly and statically linked.

**Review by asemjonovs — DISMISSED** [review]

**Comment by aaronemassey** [comment]
> The function renames look fine to me, but could they actually be overridden by other symbols before? They appear to be strongly and statically linked.

I'll clarify in the commit message that this is supposed to support changes like https://github.com/zephyrproject-rtos/zephyr/pull/46817/commits/03d1ff64bc92aa1fa6cc70fed53869009062508a where it may be changed to test_mockable.

File: subsys/emul/i2c/emul_sbs_gauge.c (hunk 2 above)
Comment (fabiobaltieri) [reply subsys/emul/i2c/emul_sbs_gauge.c]: Yeah I saw it on the CI run, one step was failing.

**Review by fabiobaltieri — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by yperess — APPROVED** [review]

**Comment by aaronemassey** [comment]
Adding DNM as there's a test I'd like to do locally real quick that looked like I was skipping. Removing momentarily.

**Comment by aaronemassey** [comment]
Removed DNM, all is well. This is ready to merge.

**Pull request merged**